src/wp-includes/global-styles-and-settings.php

```diff
@@ -353,6 +375,10 @@ function wp_add_global_styles_for_blocks() {
 			}
 		}
 	}
+
+	if ( $can_use_cached ) {
+		set_site_transient( $cache_key, $cached, HOUR_IN_SECONDS );
```

Comment:
This will cause the transient expiry time to get bumped even when the cached value isn't changing, which also results in 2 additional DB queries, so we should only update the transient if after checking whether the cached value has changed.

Reply:
Yes, you are correct. Thanks for pointing this out.

Reply:
Once https://github.com/WordPress/wordpress-develop/pull/6311/files is merged, that will be just one query.